src/main/kotlin/dev/arbjerg/ukulele/features/LeaveOnIdle.kt

```diff
@@ -0,0 +1,78 @@
+package dev.arbjerg.ukulele.features
+
+import dev.arbjerg.ukulele.audio.Player
+import org.springframework.stereotype.Service
+import dev.arbjerg.ukulele.audio.PlayerRegistry
+import dev.arbjerg.ukulele.config.BotProps
+import dev.arbjerg.ukulele.data.GuildPropertiesService
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import net.dv8tion.jda.api.entities.Guild
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.VoiceChannel
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.TaskScheduler
+import java.time.Duration
+import java.util.concurrent.ScheduledFuture
+
+@Service
+class LeaveOnIdle(
+        private val players: PlayerRegistry,
+        private val guildPropertiesService: GuildPropertiesService,
+        private val scheduler: TaskScheduler,
+        private val botProps: BotProps
+) {
+    private val log: Logger = LoggerFactory.getLogger(LeaveOnIdle::class.java)
+    private val timers: MutableMap<Guild, ScheduledFuture<*>> = HashMap()
+
+    fun onVoiceJoin(guild: Guild, channel: VoiceChannel, member: Member) {
```

Comment:
Strictly speaking, this method and its twin only really need the guild parameter. I would put log and the member check in the event handler